scripts/sagas/collection.js

```diff
@@ -184,20 +179,12 @@ function shouldProcessAttachment(serverInfo, records) {
 }
 
 export function* watchListRecords() {
-  // XXX: we can optionnaly be passed a ROUTE_LOAD_SUCCESS action payload here,
-  // we may eventually want to extract a default sort value from the provided
-  // collection data.
   while(true) { // eslint-disable-line
     const {bid, cid, sort} = yield take(COLLECTION_RECORDS_REQUEST);
     yield fork(listRecords, bid, cid, sort);
   }
 }
 
-export function* watchResetListRecords() {
-  // On each route update, reset the records loader
-  yield* takeLatest(ROUTE_LOAD_SUCCESS, watchListRecords);
```

Comment:
Note: I realized that while this works, we're not ensuring the route data are loaded prior to loading the list of records anymore, which may (or may not) introduce race conditions. Need to think about it a little more.